tests/ha/graceful_shutdown.pm

```diff
@@ -0,0 +1,50 @@
+# SUSE's openQA tests
+#
+# Copyright 2019 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: ha-cluster-bootstrap
+# Summary: shutdown a cluster gracefully and start it up again using "crm cluster" commands
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use hacluster;
+use version_utils 'is_sle';
+use utils;
+
+sub run {
+    if (is_sle('>15-sp3')) {
+        my $cluster_name = get_cluster_name;
+        my $crm_timeout = bmwqemu::scale_timeout(60);
+
+        # Waiting for the other nodes to be ready
+        barrier_wait("CLUSTER_GRACEFUL_SHUTDOWN_" . "$cluster_name");
+
+        # graceful shutdown
+        assert_script_run("crm cluster stop --all", $crm_timeout) if is_node(1);
+
+        wait_until_resources_stopped;
+
+        # confirm cluster is stopped (only needed for logging purposes)
+        validate_script_output_retry("systemctl --no-pager -l status corosync", sub { m/Stopped Corosync/i });
```

Comment:
It would be good to have some info message at least before the 'validate script output retry' so it is easier to troubleshoot and read the result. Something like "record_info('corosync')" so you can easy find outputs from commands since validate_script_output has only title 'validate_script'.

Reply:
:) implemented